app/configuration.py

```diff
@@ -74,6 +83,14 @@ def __init__(
         self.holding_register_map = {x.name: x for x in holding_registers}
         self.mqtt_settings = mqtt_settings
         self.modbus_settings = modbus_settings
+        self.env_settings = self._load_env_settings()
+
+    @classmethod
+    def _load_env_settings(cls) -> EnvSettings:
```

Comment:
I like where this is heading to but I'd like is to explore more how telegraf resolved this
https://github.com/influxdata/telegraf/blob/master/plugins/outputs/mqtt/README.md?plain=1#L53

Essentially you can still get the content from the configuration file. I'm our case, just putting this two in the yaml file.
The difference though is inside this class where we can interpret some specific symbols as environment variable.

If we read that the `site_name` is called `$SITE_NAME` we do environment evaluation. We can deal with that using a strict Allowlist or a loose one, for example, allowing any string of the format `$SOME_VAR`.

I'd start by having a strict Allowlist.

If you see a hardcoded string, you would just use that instead. No need to do environment variable.

That way, we allow this project to remain flexible and still be compatible with our other configuration

Reply:
OK, this is a much better idea - keeps all the config settings coming from a single source.